prompt_toolkit/mouse_events.py

```diff
@@ -38,9 +67,22 @@ class MouseEvent:
     :param event_type: `MouseEventType`.
     """
 
-    def __init__(self, position: Point, event_type: MouseEventType) -> None:
+    def __init__(
+        self,
+        position: Point,
+        event_type: MouseEventType,
+        button: MouseButton,
+        modifier: MouseModifier,
```

Comment:
I don't think it makes sense for ``modifier`` to be an enum. ``Set[MouseModifier]`` seems more logical.

Reply:
Set[MouseModifier] makes more semantic sense, but the reason I decided to do it this way is because Set[MouseModifier] isn't hashable - meaning you can't use Set[MouseModifier] as keys in a dictionary to branch off efficiently

Reply:
We have frozenset for that. ;)

While I merge it, I'll probably do some polishing, and use frozen sets. If that's fine for you.

Reply:
I just looked into that...yeah I think that would work well! I don't know how I missed that feature lol:)